pkg/porter/install.go

```diff
@@ -32,6 +35,14 @@ func (p *Porter) InstallBundle(opts InstallOptions) error {
 		return err
 	}
 
+	if len(opts.CredentialIdentifiers) == 0 {
```

Comment:
One critical part of this check that is missing right now is that we don't want to prompt for credentials unless credentials are needed by the bundle for this action. Otherwise we would bring up this prompt when a credential isn't required.

I don't want to compound the problems that we are already facing in #877.

For example, the hello world bundle that we have people run doesn't have credentials and this change as-is would force them to create credentials. So as a litmus test, let's make sure that the solution allows someone to run the [quickstart](https://porter.sh/quickstart/#install-from-the-registry) without modification to use credentials.